test/models/pet_test.rb

```diff
@@ -126,4 +126,38 @@ class PetTest < ActiveSupport::TestCase
       assert_not pet.is_adopted?
     end
   end
+
+  context "ransack_adopted" do
+    setup do
+      create(:pet)
+      create(:pet)
+      create(:pet, :adopted)
+    end
+
+    should "should return adopted pets" do
+      adopted_pets = Pet.ransack_adopted(true)
+      assert_equal 1, adopted_pets.count
+    end
+
+    should "should return unadopted pets" do
+      unadopted_pets = Pet.ransack_adopted(false)
+      assert_equal 2, unadopted_pets.count
+    end
+
+    # FOR DEBUGGING
+    should "should return adopted pets via Pet.ransack" do
+      adopted_pets = Pet.ransack({"ransack_adopted"=>"true"})
+
+      assert_equal Pet.ransack_adopted(true).to_sql, adopted_pets.result.to_sql
+      assert_equal 1, adopted_pets.result.count
+    end
+
+    # Fails. SQL isn't what we'd expect?
+    should "should return unadopted pets via Pet.ransack" do
+      unadopted_pets = Pet.ransack({"ransack_adopted"=>"false"})
```

Comment:
This isn't returning the SQL we'd expect. In other words, it's not calling the else condition of `def self.ransack_adopted(boolean)`. 

Perhaps class methods with conditional logic isn't supported by `ransackable_scopes` 🤔

Reply:
@khiga8 sorry, I did not get notified when you made this comment...only just saw it. 

It's currently a bit weird. The method returns a scope depending on the argument, which is OK. But, we have an `adopted` scope defined on `Pet`, but I do not see one defiend for `unadopted`. Can you try creating an `unadopted` scope and see if this filter behaves as expected?  For the unadopted scope, we will want to find Pets that do not have a Match with `type` `adoption`. All other pets should be included in the results (including those with a Match `type` of `foster`.
 
```
  def self.ransack_adopted(boolean)
    boolean ? adopted : unadopted # there is no unadopted scope on the Pet model
  end
```

Reply:
Ok I get what you're saying now I am caught up. So, we have asserted the expected behaviour of this filter in this file, right? `test/controllers/organizations/staff/pets_controller_test.rb` 

And you have tested this works as expected in the browser?

If so, I am comfortable removing these tests. I don't think we need to assert the SQL. Thoughts?

Reply:
I can check out your branch this week to see but before I can get to that can you check the pet object you create in the setup to make sure it belongs to the expected organization

Reply:
Hey @kasugaijin. The Pet object within my test all belongs to the same organization.

I've updated the tests so it's clearer what cases are failing. 

The following scenarios work as expected (as shown by the passing tests):
* `Pet.ransack_adopted(true)`
* `Pet.ransack_adopted(false)`
* `Pet.ransack({ "ransack_adopted" => "true" })

However, the following scenario fails:
* `Pet.ransack({ "ransack_adopted" => "false" })

Curious if you might know why  `Pet.ransack_adopted(false)` returns expected results, but `Pet.ransack({ "ransack_adopted" => "false" })` doesn't.

Reply:
@khiga8 I think you are running into an open issue/bug with the ransack gem. This workaround might help: https://github.com/activerecord-hackery/ransack/issues/1375#issuecomment-1674885632

Reply:
Amazing @coalest, thank you for pointing me to that 😁